web_external/js/models/DatasetModel.js

```diff
@@ -66,313 +39,153 @@ minerva.models.DatasetModel = minerva.models.MinervaModel.extend({
-                type: 'error',
-                timeout: 4000
-            });
-        }, this));
+        this.saveMinervaMetadata(mm);
+        return mm;
     },
 
-    getAllFiles: function (callback) {
-        if (!this.files) {
-            this.files = new girder.collections.FileCollection();
-            this.files.resourceName = 'item/' + this.get('_id') + '/files';
-            this.files.append = true; // Append, don't replace pages
-            this.files.on('g:changed', function () {
-                callback(this);
-            }, this).fetch();
-        } else {
-            callback(this);
-        }
+    /**
+     * Getter for the GeoJs rendering type of the dataset, initializing it if
+     * necessary by inference from the dataset's metadata, and saving
+     * the dataset's metadata if changed.
+     *
+     * @returns {'choropleth'|'geojson'|'contour'|'wms'|null} GeoJs rendering type of this dataset, will
+     * be null if no rendering type can be inferred.
+     */
+    getGeoRenderType: function () {
+        var mm = this._initGeoRender();
+        return mm.geo_render ? mm.geo_render.type : mm.geo_render;
     },
 
-    // TODO split CSV functionality out possibly to a subclass
-
-    getCSVFile: function (callback) {
-        var processCSVFile = _.bind(function () {
-            // TODO get the csv in some smarter way
-            // the file object has a mimeType, which could check for text/csv
-            // TODO better dealing with getting model from collection
-            $.ajax({
-                url: girder.apiRoot + '/file/' + this.files.models[0].id + '/download',
-                type: 'GET',
-                headers: {
-                    'Girder-Token': girder.cookie.find('girderToken'),
-                    Accept: 'text/csv; charset=utf-8',
-                    'Content-Type': 'text/csv; charset=utf-8'
-                },
-                success: _.bind(function (resp) {
-                    this.csv = Papa.parse(resp, {skipEmptyLines: true});
-                    callback();
-                }, this),
-                error: function (err) {
-                    console.error(err);
-                    girder.events.trigger('g:alert', {
-                        icon: 'cancel',
-                        text: 'Could not download dataset csv contents.',
-                        type: 'error',
-                        timeout: 4000
-                    });
-                }
-            });
-
-        }, this);
-        if (this.files) {
-            processCSVFile();
+    /**
+     * Gets the download URL for the file data needed by GeoJs to render
+     * this dataset, if one exists; initializing it if
+     * necessary by inference from the dataset's metadata, and saving
+     * the dataset's metadata if changed.
+     *
+     * @returns {String|null} Download URL for the file data, if one exists.
+     */
+    _getGeoRenderDownloadUrl: function () {
```

Comment:
I think this is a slightly different situation and can't use the girder.Model.downloadUrl.

That method is for downloading the entire resource, so in the case of an item would download all files.

In this case, I just want to download a single file, and it is a specific file related to geo rendering.